### PR TITLE
docs(gatsby): improve createPages example

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

File: packages/gatsby/src/utils/api-node-docs.js [hunk 1]
```diff
@@ -12,49 +12,51 @@ exports.resolvableExtensions = true
...
- *                 //
- *                 // The page "path" is always available as a GraphQL
- *                 // argument.
- *               },
- *             })
- *         })
+ *     }
+ *   `).then(result => {
+ *     if (result.errors) {
+ *       reject(result.errors)
```
Comment (DSchau) [inline packages/gatsby/src/utils/api-node-docs.js]:
```suggestion
  *       return Promise.reject(result.errors)
```

File: packages/gatsby/src/utils/api-node-docs.js [hunk 2]
```diff
@@ -12,49 +12,51 @@ exports.resolvableExtensions = true
...
- *           edges {
- *             node {
- *               fields {
- *                 slug
- *               }
+ *   const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
+ *   // Query for markdown nodes to use in creating pages.
+ *   // You can query for whatever data you want to create pages for e.g.
+ *   // products, portfolio items, landing pages, etc.
+ *   graphql(`
```
Comment (DSchau) [inline packages/gatsby/src/utils/api-node-docs.js]: Do we want to return here? It doesn't really matter either way, but it doesn't _hurt_

File: packages/gatsby/src/utils/api-node-docs.js [hunk 3]
```diff
@@ -12,49 +12,51 @@ exports.resolvableExtensions = true
...
+ *           // arguments to the page GraphQL query.
+ *           //
+ *           // The page "path" is always available as a GraphQL
+ *           // argument.
+ *         },
  *       })
- *     )
+ *     })
+ *
+ *     return
```
Comment (DSchau) [inline packages/gatsby/src/utils/api-node-docs.js]: We can probably just drop this? The flow ends here anyways, particularly if we return the `graphql` call above.

**Comment by KyleAMathews** [comment]
This is a similar change to @pobch's in https://github.com/gatsbyjs/gatsby/pull/10357

File: packages/gatsby/src/utils/api-node-docs.js (hunk 3 above)
Comment (KyleAMathews) [reply packages/gatsby/src/utils/api-node-docs.js]: 👍

File: packages/gatsby/src/utils/api-node-docs.js (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby/src/utils/api-node-docs.js]: Oh this should be a `throw`, yikes, didn't read this very closely.

**Comment by KyleAMathews** [comment]
Hmm now that I think about it, in the past I've avoided throwing because that meant you couldn't get to graphiql to try fixing the error.

I wonder if we should automatically catch graphql errors internally and show a nice clean error with a link to open the query as is in graphiql and never return graphql errors.

/cc @gatsbyjs/core

**Pull request merged**